app/assets/javascripts/host_edit.js

```diff
@@ -2,6 +2,47 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
 $(document).on('AddedClass', function(event, link){load_puppet_class_parameters(link)});
 $(document).on('click', '#params-tab', function() { resizeTextareas($('#params')); });
 
+LookupValueController = function(item) {
+  var self = this; //because JS' this confuses me
+  self.element = item;
+
+  self.constructor = function(){
+    self.id = self.element.data("lookupKeyId");
+    self.name = self.element.find(".lookup_key_name").html();
+    self.value = self.element.find(".value").val();
+    self.overrideTR = $("#lookup_value_for_"+self.id);
+    self.destroyer = self.overrideTR.find(".destroyer")
+    self.addEventListeners();
+  }
+
+  self.addEventListeners = function() {
+    self.element.find(".override_button").click(self.override);
+    self.overrideTR.find(".remove_override_button").click(self.removeOverride);
+  }
+
+  self.override = function() {
+    //create the overridden field
+    self.element.addClass("overridden");
+    self.overrideTR.addClass("visible");
```

Comment:
:+1: 

That's fine, I don't feel strongly enough about it to hold up these changes.